x-pack/plugins/stack_connectors/public/connector_types/slack_api/slack_connectors.tsx

```diff
@@ -33,15 +42,123 @@ const getSecretsFormSchema = (docLinks: DocLinksStart): SecretsFieldSchema[] =>
   },
 ];
 
+const getConfigFormSchemaAfterSecrets = (
+  options: EuiComboBoxOptionOption[],
+  isLoading: boolean,
+  isDisabled: boolean
+): ConfigFieldSchema[] => [
+  {
+    id: 'allowedChannels',
+    isRequired: false,
+    label: i18n.ALLOWED_CHANNELS,
+    helpText: (
+      <FormattedMessage
+        id="xpack.stackConnectors.components.slack_api.allowedChannelsText"
+        defaultMessage="By default, the connector can access all channels within the scope of the Slack app."
+      />
+    ),
+    type: 'COMBO_BOX',
+    euiFieldProps: {
+      isDisabled,
+      isLoading,
+      noSuggestions: false,
+      options,
+    },
+  },
+];
+
 const SlackActionFields: React.FC<ActionConnectorFieldsProps> = ({ readOnly, isEdit }) => {
-  const { docLinks } = useKibana().services;
+  const {
+    docLinks,
+    http,
+    notifications: { toasts },
+  } = useKibana().services;
+
+  const form = useFormContext();
+  const { setFieldValue } = form;
+  const [formData] = useFormData({ form });
+  const [authToken, setAuthToken] = useState('');
+  const [channels, setChannels] = useState<EuiComboBoxOptionOption[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const abortCtrlRef = useRef(new AbortController());
+  const isMounted = useRef(false);
+
+  async function fetchChannels(newAuthToken: string) {
+    setIsLoading(true);
+    setChannels([]);
+    setFieldValue('config.allowedChannels', []);
+    isMounted.current = true;
+    abortCtrlRef.current.abort();
+    abortCtrlRef.current = new AbortController();
+
+    try {
+      const res = await http.post<GetChannelsResponse>(
+        `${INTERNAL_BASE_STACK_CONNECTORS_API_PATH}/_slack_api/channels`,
```

Comment:
It's a pity we cannot use the sub-actions framework.

Reply:
if you do not have an id, it won't work, we can always later on refactor to allow the query without an id

Reply:
oh yeah, good point.